src/lib/utils.ts

```diff
@@ -14,6 +14,13 @@ export function isArray(val: any): val is unknown[] {
   return Array.isArray(val)
 }
 
+/**
+ * An alternative to `Object.keys()` that returns an array of keys with the correct type.
+ */
+export function typedKeys<T extends object>(obj: T): Array<keyof T> {
+  return Object.keys(obj) as Array<keyof T>
+}
```

Comment:
@jtran lmk if there's a better way of doing this.

Reply:
This is actually not safe because of TS's structural subtyping and how at runtime, you can extend a JS object with whatever keys you want. `Object.keys()` will return them all, and that won't be an `Array<keyof T>`. It would have additional keys in the array.

I get where you're coming from though. Most of the time, this is nicer. So not sure what to say. Might want to rename it to `unsafeTypedKeys` or at least leave a note in the doc comments for anyone who might use this that it only works under certain conditions.

Reply:
Good call.

I would hope the run time issue is not a problem for us, since that would require other bad typescirpt, but the structural subtyping is a good point.

I changed the name and expanded to doc string.